lib/matplotlib/hatch.py

```diff
@@ -186,14 +185,11 @@ def _validate_hatch_pattern(hatch):
         if invalids:
             valid = ''.join(sorted(valid_hatch_patterns))
             invalids = ''.join(sorted(invalids))
-            _api.warn_deprecated(
-                '3.4',
-                removal='3.9',  # one release after custom hatches (#20690)
-                message=f'hatch must consist of a string of "{valid}" or '
-                        'None, but found the following invalid values '
-                        f'"{invalids}". Passing invalid values is deprecated '
-                        'since %(since)s and will become an error %(removal)s.'
-            )
+            message = f"""Unknown hatch symbol(s): {invalids}.
+                          Hatch must consist of a string of {valid}"""
+            raise ValueError(message)
+    else:
+        raise ValueError("Hatch pattern must be a string")
```

Comment:
The referenced issue RE: `custom hatches` (#20690) is still open...

That said, it has been warning for a while now, and there is no real motion on that issue, I guess we may be able to expire this deprecation even without that.

Also the original wording included the valid value of `None`, which this wording omits.

Thoughts @timhoffm?

Reply:
I'm sorry, I had already started working on the project before and mistakenly picked an older version of the code to make changes (from last year). I will rectify this and restore all the functions that I accidentally removed. I will resubmit my pull request. I'm new to the project and hope to be able to contribute.

Reply:
You can update your pull request by pushing additional commits (or force-pushing a rebased branch) to the same branch, please do not open new pull requests, this helps keep the conversation in one place so we can more easily see that things are addressed.